src/LSST/system/TelemetryItem.h

```diff
@@ -114,115 +132,297 @@ class TelemetryItem {
+        }
+        return false;
+    }
 
     /// Return true if this item and `other` have the same id and values.
-    bool compareItem(TelemetryItem const& other) const override;
+    bool compareItem(TelemetryItem const& other) const override {
+        try {
+            TItemSimple<ST> const& otherItem = dynamic_cast<TItemSimple<ST> const&>(other);
+            return (getId() == otherItem.getId() && _val == otherItem._val);
+        } catch (std::bad_cast const& ex) {
+            return false;
+        }
+    }
 
 private:
-    std::atomic<double> _val;
+    std::atomic<ST> _val; ///< Vector containing the double values with length _size.
+    mutable std::mutex _mtx; ///< Protects `_val`.
 };
 
-/// common elements between powerStatus and powerStatusRaw
+
+/// This class is used to store a specific telemetry item with a double value.
 /// Unit tests in tests/test_TelemetryCom
-class TItemPowerStatusBase : public TelemetryItem {
+class TItemDouble : public TItemSimple<double> {
 public:
-    TItemPowerStatusBase(std::string const& id) : TelemetryItem(id) {}
+    using Ptr = std::shared_ptr<TItemDouble>;
 
-    virtual ~TItemPowerStatusBase() = default;
+    /// Create a shared pointer instance of TItemDouble using `id`, and `defaultVal`, and
+    /// insert it into the map `itMap`.
+    /// @throws TelemetryException if the new object cannot be inserted into the list.
+    static Ptr create(std::string const& id, TelemetryItemMap* tiMap, double defaultVal = 0.0) {
+        Ptr newItem = Ptr(new TItemDouble(id, defaultVal));
+        insert(tiMap, newItem);
+        return newItem;
+    }
 
-    /// Set `_motorVoltage` to `val`.
-    void setMotorVoltage(double val) { _motorVoltage->setVal(val); }
+        /// Create a `TItemDouble` object with identifier `id` and option value `defaultVal`.
+    TItemDouble(std::string const& id, double defaultVal = 0.0) : TItemSimple(id, defaultVal) {}
 
-    /// Set `_motorCurrent` to `val`.
-    void setMotorCurrent(double val) { _motorCurrent->setVal(val); }
+    TItemDouble() = delete;
 
-    /// Set `_commVoltage` to `val`.
-    void setCommVoltage(double val) { _commVoltage->setVal(val); }
+    virtual ~TItemDouble() {}
+};
 
-    /// Set `_commCurrent` to `val`.
-    void setCommCurrent(double val) { _commCurrent->setVal(val); }
+/// This class is used to store a specific telemetry item with a bool value.
+/// Unit tests in tests/test_TelemetryCom
+class TItemBoolean : public TItemSimple<bool> {
+public:
+    using Ptr = std::shared_ptr<TItemBoolean>;
 
-    /// Return the value of `_motorVoltage`.
-    double getMotorVoltage() { return _motorVoltage->getVal(); }
+    /// Create a shared pointer instance of TItemBoolean using `id`, and `defaultVal`, and
+    /// insert it into the map `itMap`.
+    /// @throws TelemetryException if the new object cannot be inserted into the list.
+    static Ptr create(std::string const& id, TelemetryItemMap* tiMap, bool defaultVal = 0.0) {
```

Comment:
If the data type of `defaultVal` is `bool`, why the default value is not `true/false`?

Reply:
fixed